pkg/api/dbaas/mongodbatlasinventory_types.go

```diff
@@ -0,0 +1,160 @@
+	// provider name specified in the provider’s operator in its registration ConfigMap.
+	Provider DatabaseProvider `json:"provider"`
+
+	// The properties that will be copied into the provider’s inventory Spec
+	MongoDBAtlasInventoryProviderSpec `json:"providerSpec"`
+}
+
+// MongoDBAtlasInventoryProviderSpec The Inventory Spec to be used by provider operators
+type MongoDBAtlasInventoryProviderSpec struct {
+	// The Secret containing the provider-specific connection credentials to use with its API
+	// endpoint.  The format of the Secret is specified in the provider’s operator in its registration
+	// ConfigMap (credentials_fields key).  It is recommended to place the Secret in a namespace
+	// with limited accessibility.
+	CredentialsRef *NamespacedName `json:"credentialsRef"`
+}
+
+type NamespacedName struct {
+	// The namespace where object of known type is stored
+	Namespace string `json:"namespace,omitempty"`
+
+	// The name for object of known type
+	Name string `json:"name,omitempty"`
+}
+
+type DatabaseProvider struct {
+	Name string `json:"name"`
+}
+
+// MongoDBAtlasInventoryStatus defines the observed state of MongoDBAtlasInventory
+type MongoDBAtlasInventoryStatus struct {
+	status.Common `json:",inline"`
+
+	// Type used by the Service Binding Operator - e.g., MongoDB, Postgres
+	ServiceBindingType string `json:"type"`
+
+	// Provider used by the Service Binding Operator
+	ServiceBindingProvider string `json:"provider"`
+
+	// A list of instances returned from querying the DB provider
+	Instances []Instance `json:"instances,omitempty"`
+}
+
+type Instance struct {
+	// A provider-specific identifier for this instance in the database service.  It may contain one or
+	// more pieces of information used by the provider operator to identify the instance on the
+	// database service.
+	InstanceID string `json:"instanceID"`
+
+	// The name of this instance in the database service
+	Name string `json:"name,omitempty"`
+
+	// Any other provider-specific information related to this instance
+	InstanceInfo InstanceInfo `json:"extraInfo,omitempty"`
+}
+
+type InstanceInfo struct {
+	CloudProvider    string `json:"providerName"`
+	CloudRegion      string `json:"regionName"`
+	ProjectID        string `json:"projectID"`
+	ProjectName      string `json:"projectName"`
+	InstanceSizeName string `json:"instanceSizeName"`
+	// ConnectionString is for your applications use to connect to this cluster.
+	ConnectionString string `json:"connectionString,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +groupName:=dbaas.redhat.com
+// +versionName:=v1alpha1
+
+// MongoDBAtlasInventory is the Schema for the MongoDBAtlasInventory API
+type MongoDBAtlasInventory struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MongoDBAtlasInventorySpec   `json:"spec,omitempty"`
```

Comment:
The Spec and status will be referenced from DBass Operator, like https://github.com/CrunchyData/crunchy-bridge-operator/blob/52bd64e43f290d75631957f1d36ed3780fba0042/apis/dbaas.redhat.com/v1alpha1/crunchybridgeinventory_types.go#L35

Reply:
cc: @bmozaffa

Reply:
@jianrongzhang89 InstanceInfo is a map, so you can return any arbitrary information in there:

https://github.com/RHEcosystemAppEng/dbaas-operator/blob/main/api/v1alpha1/dbaasprovider.go#L86

Reply:
@bmozaffa I see. Thanks.

Reply:
@bmozaffa @redhatHameed  I  reimplemented the code using the data structures from dbaas-operator directly. Please take a look again.